src/Microsoft.Azure.SignalR.Common/ServiceConnections/StrongServiceConnectionContainer.cs

```diff
@@ -33,22 +33,20 @@ public override Task HandlePingAsync(PingMessage pingMessage)
             return Task.CompletedTask;
         }
 
-        public override Task StopAsync()
+        public override async Task StopAsync()
         {
             var task = base.StopAsync();
-            return Task.WhenAll(
+            await Task.WhenAll(
                 task,
                 Task.WhenAll(_onDemandServiceConnections.Select(c => c.StopAsync()))
             );
         }
 
-        public override Task ShutdownAsync(TimeSpan timeout)
+        public override Task OfflineAsync()
         {
-            var task = base.StopAsync();
-            return Task.WhenAll(
-                task,
-                Task.CompletedTask // TODO
-            );
+            var task1 = base.OfflineAsync();

```

Comment:
more meaningful naming

Reply:
I think everyone could understand that I just want to combine these 2 tasks and waiting for all of them to be completed.